README.md

```diff
@@ -41,96 +73,185 @@ pod "KlaviyoSwift"
+The SDK **should** be initialized before any other Klaviyo SDK methods are called.
+
+## Profile Identification
+The SDK provides methods to identify your users as Klaviyo profiles via the [Create Client Profile API](https://developers.klaviyo.com/en/reference/create_client_profile).
+A profile can be identified by any combination of the following:
+
+* External ID: A unique identifier used by customers to associate Klaviyo profiles with profiles in an external system, such as a point-of-sale system. Format varies based on the external system.
+* Individual's email address
+* Individual's phone number in [E.164 format](https://help.klaviyo.com/hc/en-us/articles/360046055671#h_01HE5ZYJEAHZKY6WZW7BAD36BG)
+
+These above Identifiers are persisted to local storage so that the SDK can keep track of the current user/profile for you when you make event requests or want to set a push token etc.
+
+Profile identifiers can be set all at once or individually. Either way, the SDK will group and batch API calls to improve performance.
+
+The following code demonstrates how to set profile identifiers:
 
 ```swift
-KlaviyoSDK().initialize(with: "YOUR_KLAVIYO_PUBLIC_API_KEY")
+// organization, title, image, location and additional properties (dictionary) can also be set using the below constructor
+let profile = Profile(email: "junior@blob.com",  firstName: "Blob",  lastName: "Jr.")
+KlaviyoSDK().set(profile: profile)
+
+// or setting individual properties
+KlaviyoSDK().set(profileAttribute: .firstName, value: "Blob")
+KlaviyoSDK().set(profileAttribute: .lastName, value: "Jr.")
 ```
 
-3. Begin tracking events anywhere within your application by calling the `create(event:)` method in the relevant location.
+### Reset Profile
+To start a new profile altogether (e.g. if a user logs out) either call `KlaviyoSDK().resetProfile()` to clear the currently tracked profile identifiers,
+or use `KlaviyoSDK().set(profile: profile)` to overwrite it with a new profile object.
+<TODO: Should we talk about what happens to a push token here?>
 
 ```swift
-let event = Event(name: .AddedToCartMetric, properties: [
-    "Total Price": 10.99,
-    "Items": ["Hot Dog", "Fries", "Shake"]
-], identifiers: .init(email: "junior@blob.com"),
-profile: [
-    "$first_name": "Blob",
-    "$last_name": "Jr"
-], value: 10.99)
+// start a profile for Blob Jr.
+let profile = Profile(email: "junior@blob.com",  firstName: "Blob",  lastName: "Jr.")
+KlaviyoSDK().set(profile: profile)
 
-KlaviyoSDK().create(event: event)
+// stop tracking Blob Jr.
+KlaviyoSDK().resetProfile()
+
+// start a profile for Robin Hood
+let profile = Profile(email: "junior@blob.com",  firstName: "Robin",  lastName: "Hood")
+KlaviyoSDK().set(profile: profile)
 ```
+### Anonymous Tracking Notice
 
-### Arguments
+By default, Klaviyo will begin tracking unidentified users in your app when a push token is set or we receive an event to track.
```

Comment:
If you do not identify an app user as outlined above, Klaviyo will automatically create an anonymous profile for unidentified users when you set a push token or track an event. Klaviyo identifies these profiles through an anonymous ID, a randomly generated UUID.